sys/linux/dev_vga_arbiter.txt

```diff
@@ -0,0 +1,44 @@
+# Copyright 2020 syzkaller project authors. All rights reserved.
+# Use of this source code is governed by Apache 2 LICENSE that can be found in the LICENSE file.
+
+resource fd_vga_arbiter[fd]
+openat$vga_arbiter(fd const[AT_FDCWD], file ptr[in, string["/dev/vga_arbiter"]], flags flags[open_flags], mode const[0]) fd_vga_arbiter
+
+write$vga_arbiter(fd fd_vga_arbiter, data ptr[in, data_vga_arbiter], len bytesize[data])
+
+data_vga_arbiter [
+	target_default	string["target default"]
+	unlock_all	string["unlock all"]
+	target		vga_arbiter_target_cmd
+	other		vga_arbiter_io_state_cmd
+] [varlen]
+
+vga_arbiter_target_cmd {
+	prefix	stringnoz["target "]
+	card_ID	vga_arbiter_card_ID
+	end	const[0, int8]
+} [packed]
+
+# We cannot use fmt for the numbers here or else the result string will be too long
+vga_arbiter_card_ID {
+	prefix	stringnoz["PCI:"]
+	domain	stringnoz[vga_arbiter_pci_numbers]
+	colon1	stringnoz[":"]
+	bus	stringnoz[vga_arbiter_pci_numbers]
+	colon2	stringnoz[":"]
+	dev	stringnoz[vga_arbiter_pci_numbers]
+	dot	stringnoz["."]
+	fn	stringnoz[vga_arbiter_pci_numbers]
+} [packed]
+
+vga_arbiter_pci_numbers = "0", "1", "2", "3", "4", "5"
```

Comment:
Technically, any hex number is valid here.  But I suppose it's unlikely to have more than a couple dozen PCI devices.  `lspci` on my laptop gives 17 devices, so maybe we should increase the range here.

Reply:
Ok, I will increase the max number.